File: frontendapp/lib/models/booking.dart
class Booking {
  final int id;
  final int userId;
  final int showtimeId;
  final String bookingDate;
  final double totalPrice;
  final String paymentMethod;
  final String paymentStatus;
  final bool isActive;

  Booking({
    required this.id,
    required this.userId,
    required this.showtimeId,
    required this.bookingDate,
    required this.totalPrice,
    required this.paymentMethod,
    required this.paymentStatus,
    required this.isActive,
  });

  factory Booking.fromJson(Map<String, dynamic> json) => Booking(
    id: json['id'],
    userId: json['userId'],
    showtimeId: json['showTimeId'],
    bookingDate: json['bookingdate'],
    totalPrice: json['totalprice'].toDouble(),
    paymentMethod: json['paymentmethod'],
    paymentStatus: json['paymentstatus'],
    isActive: json['isactive'],
  );
}

File: backend/bookingmovieticketapp/app/Models/Booking.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Booking extends Model
{
    use HasFactory;
    protected $table = "bookings";
    protected $primaryKey = "id";
    public $timestamps = false;
    protected $fillable = [
        'userid',
        'showtimeid',
        'bookingdate',
        'totalprice',
        'paymentmethod',
        'paymentstatus',
        'isactive'
    ];
    protected $casts = [
        'bookingdate'=> 'datetime',
        'isactive'=> 'boolean',
        'totalprice'=>'integer'
    ];

    public function user(){
        return $this->belongsTo(User::class,'userid');
    }
    
    public function showtime(){
        return $this->belongsTo(ShowTime::class,'showtimeid');
    }

    public function bookingdetail(){
    return $this->hasMany(BookingDetail::class,'bookingid');
    }

    public function payment(){
    return $this->hasOne(Payment::class,'bookingid');
    }
}
